results.js

```diff
@@ -9,50 +9,80 @@ var config = {
 firebase.initializeApp(config);
 
 var storage = firebase.app().storage();
+var storage_ref = storage.ref();
 var db = firebase.firestore();
 
 function initialize(){
     var doctype = document.getElementById('doctype').value;
     var province = document.getElementById('province').value;
 
-    get_storage_bucket(doctype, province);
+    get_documents(doctype, province);
 }
 
-function get_storage_bucket(doctype, province){
-    var storage_url = 'gs://cameroonsite-5f7e2.appspot.com/';
+function get_documents(doctype, province){
+    var url = 'counties';
+    var documents_array = [];
+    var doc_url_array = [];
+
+    if(province !== 'all') {
+        url += '/'+province;
+    }
+
+    var docRef = db.doc(url);
+
+    docRef.get().then( function(doc) {
+        if( doctype !== 'Show All Documents') {
+            if(doctype === 'Audio Recordings'){
+                doctype = 'Recordings';
+            }
+            documents_array = doc.get(doctype);
+            documents_array.forEach(function(document) {
+               storage_ref.child(document).getDownloadURL().then(function(url) {
+                 doc_url_array.push(url);
+                 document.getElementById('documents').innerHTML = doc_url_array[0];
+                 var xhr = new XMLHttpRequest();
+                 xhr.responseType = 'blob';
+                 xhr.onload = function(event) {
+                   var blob = xhr.response;
```

Comment:
What's the purpose of the blob variable? Doesn't look like it's being used anywhere. My recommendation is that you write an empty function passing the XHTTP request in. UI can populate that function with whatever UI linkage needs to be done to render the image. If you want to test the image fetch live, try adding an image tag somewhere in the HTML page so that the images just get dumped to the page.